python/lsst/ip/isr/isrTaskLSST.py

```diff
@@ -0,0 +1,1219 @@
+        if config.doLinearize is not True:
+            self.prerequisiteInputs.remove("linearizer")
+        if config.usePtcGains is not True and config.usePtcReadNoise is not True:
+            self.prerequisiteInputs.remove("ptc")
+        if config.doCrosstalk is not True:
+            self.prerequisiteInputs.remove("crosstalkSources")
+            self.prerequisiteInputs.remove("crosstalk")
+        if config.doDefect is not True:
+            self.prerequisiteInputs.remove("defects")
+        if config.doBrighterFatter is not True:
+            self.prerequisiteInputs.remove("bfKernel")
+        if config.doDark is not True:
+            self.prerequisiteInputs.remove("dark")
+
+        if config.doWrite is not True:
+            self.outputs.remove("outputExposure")
+            self.outputs.remove("preInterpExposure")
+
+        if config.doSaveInterpPixels is not True:
+            self.outputs.remove("preInterpExposure")
+
+        if config.doCalculateStatistics is not True:
+            self.outputs.remove("outputStatistics")
+
+
+class IsrTaskLSSTConfig(pipeBase.PipelineTaskConfig,
+                        pipelineConnections=IsrTaskLSSTConnections):
+    """Configuration parameters for IsrTaskLSST.
+
+    Items are grouped in the order in which they are executed by the task.
+    """
+    datasetType = pexConfig.Field(
+        dtype=str,
+        doc="Dataset type for input data; users will typically leave this alone, "
+        "but camera-specific ISR tasks will override it.",
+        default="raw",
+    )
+    expectWcs = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="Expect input science images to have a WCS (set False for e.g. spectrographs)."
+    )
+    qa = pexConfig.ConfigField(
+        dtype=isrQa.IsrQaConfig,
+        doc="QA related configuration options.",
+    )
+    doHeaderProvenance = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="Write calibration identifiers into output exposure header.",
+    )
+
+    # Differential non-linearity correction.
+    doDiffNonLinearCorrection = pexConfig.Field(
+        dtype=bool,
+        doc="Do differential non-linearity correction?",
+        default=True,
+    )
+
+    doOverscan = pexConfig.Field(
+        dtype=bool,
+        doc="Do overscan subtraction?",
+        default=True,
+    )
+    overscan = pexConfig.ConfigurableField(
+        target=OverscanCorrectionTask,
+        doc="Overscan subtraction task for image segments.",
+    )
+
+    # Amplifier to CCD assembly configuration.
+    doAssembleCcd = pexConfig.Field(
+        dtype=bool,
+        default=True,
+        doc="Assemble amp-level exposures into a ccd-level exposure?"
+    )
+    assembleCcd = pexConfig.ConfigurableField(
+        target=AssembleCcdTask,
+        doc="CCD assembly task.",
+    )
+
+    # Combine snaps.
+    doSnapCombine = pexConfig.Field(
+        dtype=bool,
+        doc="Combine snaps?",
+        default=False,
+    )
```

Comment:
I think actually remove this; I think it's not going to be part of the isrTask if we have to do it, but a separate task.